src/cluster.c

```diff
@@ -1105,45 +1155,29 @@ getNodeByQuery(client *c, struct serverCommand *cmd, robj **argv, int argc, int
              * NODE <node-id>. */
             int flags = LOOKUP_NOTOUCH | LOOKUP_NOSTATS | LOOKUP_NONOTIFY | LOOKUP_NOEXPIRE;
             if ((migrating_slot || importing_slot) && !pubsubshard_included) {
-                if (lookupKeyReadWithFlags(&server.db[0], thiskey, flags) == NULL)
+                if (lookupKeyReadWithFlags(&server.db[0], this_key, flags) == NULL) {
                     missing_keys++;
-                else
+                } else {
                     existing_keys++;
+                }
             }
         }
         getKeysFreeResult(&result);
     }
 
     /* No key at all in command? then we can serve the request
      * without redirections or errors in all the cases. */
-    if (n == NULL) return myself;
-
-    /* Cluster is globally down but we got keys? We only serve the request
-     * if it is a read command and when allow_reads_when_down is enabled. */
-    if (!isClusterHealthy()) {
-        if (pubsubshard_included) {
-            if (!server.cluster_allow_pubsubshard_when_down) {
-                if (error_code) *error_code = CLUSTER_REDIR_DOWN_STATE;
-                return NULL;
-            }
-        } else if (!server.cluster_allow_reads_when_down) {
-            /* The cluster is configured to block commands when the
-             * cluster is down. */
-            if (error_code) *error_code = CLUSTER_REDIR_DOWN_STATE;
-            return NULL;
-        } else if (cmd_flags & CMD_WRITE) {
-            /* The cluster is configured to allow read only commands */
-            if (error_code) *error_code = CLUSTER_REDIR_DOWN_RO_STATE;
-            return NULL;
-        } else {
-            /* Fall through and allow the command to be executed:
-             * this happens when server.cluster_allow_reads_when_down is
-             * true and the command is not a write command */
-        }
-    }
+    if (first_node == NULL) return myself;
 
     /* Return the hashslot by reference. */
-    if (hashslot) *hashslot = slot;
+    if (hashslot) *hashslot = first_slot;
```

Comment:
Slot statistics are updated based off of this information, we will bias towards the first_slot.